lib/extensions.js

```diff
@@ -359,6 +366,29 @@ function getVersionInfo(binding) {
   ].join(eol);
 }
 
+/**
+ * Gets the platform variant, currently either an empty string or 'musl' for Linux/musl platforms.
+ *
+ * @api public
+ */
+
+function getPlatformVariant() {
+  if (process.platform !== 'linux') {
+    return '';
+  }
+  var contents = '';
+  try {
+    contents = fs.readFileSync(process.execPath);
+    if (!contents.indexOf) {
```

Comment:
and performance difference on node 7.0
```
> console.time('find');fs.readFileSync(process.execPath).indexOf('libc.musl-x86_64.so.1') !== -1;console.timeEnd('find') 
find: 15.800ms

> console.time('find');fs.readFileSync(process.execPath).toString().indexOf('libc.musl-x86_64.so.1') !== -1;console.timeEnd('find')
find: 325.538ms
```

Also I think toString will use additional memory like 20M binary x UTF16 = 40M, If I am not wrong.